nautobot_circuit_maintenance/handle_notifications/sources.py

```diff
@@ -193,9 +215,53 @@ def receive_notifications(
         self.close_session()
         return received_notifications
 
+    def validate_providers(self, logger: Job, notification_source: NotificationSource, since_txt: str) -> bool:
+        """Method to validate that the NotificartionSource has attached Providers.
+
+        Args:
+            logger (Job): Job to use its logger
+            notification_source (NotificationSource): Notification Source to validate providers
+            since_txt (str): Date string to be used to log
+
+        Returns:
+            bool: True if there are relevant providers attached or False otherwise
+        """
+        providers_with_email = []
+        providers_without_email = []
+        if not notification_source.providers.all():
+            logger.log_warning(
+                message="Skipping this email account no providers were defined.",
+            )
+            return False
+
+        for provider in notification_source.providers.all():
+            for custom_field, value in provider.get_custom_fields().items():
+                if custom_field.name == "emails_circuit_maintenances" and value:
+                    self._emails_to_fetch.extend(value.split(","))
+                    providers_with_email.append(provider.name)
+                else:
+                    providers_without_email.append(provider.name)
+
+        if providers_without_email:
+            logger.log_warning(
+                message=f"Skipping {', '.join(providers_without_email)} because these providers has no email configured.",
+            )
+
+        if not providers_with_email:
+            logger.log_warning(
+                message=f"Skipping this email account because none of the providers ({', '.join(providers_with_email)}) have at least one email defined.",
```

Comment:
`providers_with_email` is empty in this case so it doesn't need to be embedded in the log message
```suggestion
                message=f"Skipping this email account because none of the providers have at least one email defined.",
```